### ISSUE TITLE
no-constructor-attributes error when manipulating the Shadow DOM

### ISSUE THREAD

**Issue by noelmace** [43081j/eslint-plugin-wc#46]
Using `this.shadowRoot.append()` in a WC constructor leads to a _[wc/no-constructor-attributes](https://github.com/43081j/eslint-plugin-wc/blob/master/docs/rules/no-constructor-attributes.md)_ error.

```js
class Demo extends HTMLElement {
  constructor() {
    super();
    this.attachShadow({ mode: "open" });

    // ERROR: Attributes must not be interacted with in the constructor as the element may not be ready yet.
    // eslintwc/no-constructor-attributes
    this.shadowRoot.append(
      p(button("previous", "<"), button("next", ">"))
    );
  }
}
```

Yet, the [standard](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-element-conformance) only restricts manipulations of the **light** DOM, and manipulating the Shadow DOM doesn't lead to any runtime error.

IMO, this is a false positive.

**Comment by 43081j** [comment]
i had a read through the spec again, it is pretty vague around mutating children in the constructor. so i looked up the examples by the polymer team back when they documented v1 of the spec, and their example does in fact create a shadow dom tree in the constructor. so it seem you're right, good catch.

mutating the shadow dom's children = fine
mutating the light dom's children = not fine

ill see if i can look into fixing it this weekend unless you want to have a stab at it

**Comment by noelmace** [comment]
And ... done :smiley: Thanks
